### PR TITLE
Python 3.11

### PR DESCRIPTION
**Pull request by airportyh**
## Why?

Want to support python 3.11. But we had some API incompatibilities with the new Python.

## Changes

1. Made custom changes to pry_python311.c and pry_python311.h for 3.11.
2. Remove some stuff from flake to reduce development time
3. Only change to pry_python311.c vs pry_python3.c: renamed `_PyObject_CallNoArg` to `PyObject_CallNoArgs`.

There are still 2 warnings: 
1. `Py_DEPRECATED(3.11) PyAPI_FUNC(void) Py_SetProgramName(const wchar_t *);`
2. `Py_DEPRECATED(3.11) PyAPI_FUNC(void) PySys_SetArgvEx(int, wchar_t **, int)`

after trying for a few hours I decided to give up for now. Main issue:

`PySys_SetArgvEx` has[ logic that updates the python `sys.path`](https://github.com/python/cpython/blob/b86ce91bfeeb0437a38ee6b59fb86d5d37d529ee/Python/sysmodule.c#L3758) and I found replicating that logic to be hard. You'd have to make sure the main file's dir is added to the sys.path, plus anything in PYTHONPATH, plus Python's home directory.

## Testing

1. cli args should work
5. -i flag should work
6. should be able to import a library from a sibling `.py` file

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -4,16 +4,7 @@ jobs:
...
-          fingerprints:
-            - "93:22:9e:64:86:b2:d7:23:41:ab:6f:e7:1a:02:35:a8"
-      - run: apk add --no-cache --no-progress make git openssh
-      - run:
-          name: Keyscan Github (HACK) #https://discuss.circleci.com/t/known-hosts-in-circle-2-0/18544
-          command: ssh-keyscan -H github.com >> ~/.ssh/known_hosts
-      - checkout
-      - setup_remote_docker
-      - run: make test-image
+      - run: echo skip
```
Comment (cdmistman) [inline .circleci/config.yml:7]: can you delete this job instead?

File: .gitignore [hunk 2]
```diff
@@ -5,3 +5,4 @@ deps.lst
 prybar.tar.gz
 generated_*.go
 result*
+__pycache__
```
Comment (cdmistman) [inline .gitignore:8]: nit: newline please

File: flake.nix [hunk 3]
```diff
@@ -98,17 +112,18 @@
...
             pkgs.readline
-            clojureWithCP
-            pkgs.jdk11_headless
-            julia
+            # clojureWithCP
+            # pkgs.jdk11_headless
+            # julia
             pkgs.zlib
-            pkgs.ruby
+            # pkgs.ruby
```
Comment (cdmistman) [inline flake.nix:126]: can you delete these instead?

**Review by cdmistman — APPROVED** [review]
lgtm

image

File: .circleci/config.yml (hunk 1 above)
Comment (airportyh) [reply .circleci/config.yml:7]: If I did, the PR gets pending indefinitely. Maybe I need perms.

File: flake.nix (hunk 3 above)
Comment (airportyh) [reply flake.nix]: :+1:

File: .gitignore (hunk 2 above)
Comment (airportyh) [reply .gitignore]: :+1:

**Pull request merged**